### ISSUE TITLE
markdown white until save chadrc

### ISSUE THREAD

**Issue by CoreyCole** [siduck76/NvChad#2765]
**Describe the bug**
when I first load up nvim on a markdown file, all the colors are white. I get the colors to pop in by opening up my chadrc.lua and saving it, then the hot reload fixes the colors.

**To Reproduce**
- this problem started when I moved from the 2.0 config to the 2.5 lazy style
- chadrc.lua
```lua
--@type ChdrcConfig
local M = {}
M.ui = {
  statusline = {
    theme = "vscode",
    order = {
      "mode",
      "file",
      "dir",
      "diagnostics",
      "git",
      "%=",
      "lsp_msg",
      "%=",
      "lsp",
      "cursor",
      "cwd",
    },
    modules = {
      -- show open file's directory
      dir = function()
        return "(" .. vim.fn.expand "%:p:h:t" .. ")"
      end,
    },
  },
  theme = "vscode_dark",
  changed_themes = {
    vscode_dark = {
      base_16 = {
        base00 = "#000000",
        base01 = "#262626",
        base02 = "#303030",
        base03 = "#3C3C3C",
        base04 = "#464646",
        base05 = "#D4D4D4",
        base06 = "#E9E9E9",
        base07 = "#FFFFFF",
        base08 = "#D16969",
        base09 = "#B5CEA8",
        base0A = "#05ad97",
        base0B = "#BD8D78",
        base0C = "#9CDCFE",
        base0D = "#DCDCAA",
        base0E = "#C586C0",
        base0F = "#E9E9E9",
        -- base00 = "#000000",
        -- base01 = "#262626",
        -- base02 = "#303030",
        -- base03 = "#3C3C3C",
        -- base04 = "#464646",
        -- base05 = "#D4D4D4",
        -- base06 = "#E9E9E9",
        -- base07 = "#FFFFFF",
        -- base08 = "#D16969",
        -- base09 = "#f6f2e1",
        -- base0A = "#007acc",
        -- base0B = "#BD8D78",
        -- base0C = "#9CDCFE",
        -- base0D = "#aef8fc",
        -- base0E = "#C586C0",
        -- base0F = "#E9E9E9",
      },
      base_30 = {
        white = "#dee1e6",
        darker_black = "#000000",
        black = "#000000", --  nvim bg
        black2 = "#000000",
        one_bg = "#282828",
        one_bg2 = "#313131",
        one_bg3 = "#3a3a3a",
        grey = "#444444",
        grey_fg = "#4e4e4e",
        grey_fg2 = "#585858",
        light_grey = "#626262",
        red = "#D16969",
        baby_pink = "#ea696f",
        pink = "#bb7cb6",
        line = "#2e2e2e", -- for lines like vertsplit
        green = "#B5CEA8",
        green1 = "#4EC994",
        vibrant_green = "#bfd8b2",
        blue = "#569CD6",
        nord_blue = "#60a6e0",
        yellow = "#D7BA7D",
        sun = "#e1c487",
        purple = "#c68aee",
        dark_purple = "#b77bdf",
        teal = "#4294D6",
        orange = "#d3967d",
        cyan = "#9CDCFE",
        statusline_bg = "#000000",
        lightbg = "#303030",
        pmenu_bg = "#60a6e0",
        folder_bg = "#7A8A92",
      },
    },
  },
}

return M
```
- init.lua
```lua
vim.g.base46_cache = vim.fn.stdpath "data" .. "/nvchad/base46/"
vim.g.mapleader = " "

-- bootstrap lazy and all plugins
local lazypath = vim.fn.stdpath "data" .. "/lazy/lazy.nvim"

if not vim.loop.fs_stat(lazypath) then
  local repo = "https://github.com/folke/lazy.nvim.git"
  vim.fn.system { "git", "clone", "--filter=blob:none", repo, "--branch=stable", lazypath }
end

vim.opt.rtp:prepend(lazypath)

local lazy_config = require "configs.lazy"

-- load plugins
require("lazy").setup({
  {
    "NvChad/NvChad",
    lazy = false,
    branch = "v2.5",
    import = "nvchad.plugins",
    config = function()
      require "options"
    end,
  },

  { import = "plugins" },
}, lazy_config)

-- load theme
dofile(vim.g.base46_cache .. "defaults")
dofile(vim.g.base46_cache .. "statusline")

require "nvchad.autocmds"

vim.schedule(function()
  require "mappings"
end)
```

**Expected behavior**
the colors load when I first open the file with neovim

**Desktop (please complete the following information):**
 - Operating System: macos
 - Terminal: wezterm
 - Version of Neovim 0.9.5

**Screenshots**

https://github.com/NvChad/NvChad/assets/4467106/2556ac4f-a7d2-4081-a759-3cadbeea360c

**Comment by siduck** [comment]
Unable to reproduce with vanilla nvchad, hence not a nvchad issue

https://0x0.st/XzmW.mp4


I hope you didnt override our treesitter's config function, if  you did then do know to add that dofile function which loads treesitter hlgroups

**Comment by CoreyCole** [comment]
Sorry this is not an nvchad issue, my bad. I have the dotfile call in the config function. (I learned about that from some other issues) Thank you for the help bro you are awesome.

- configs/treesitter.lua
```lua
--@type TreesitterOptions
local options = {
  ensure_installed = "all",
  highlight = {
    enable = true,
    use_languagetree = true,
  },
  indent = { enable = true },
  autotag = { enable = true },
  context_commentstring = {
    enable = true,
    config = {
      javascript = {
        __default = "// %s",
        jsx_element = "{/* %s */}",
        jsx_fragment = "{/* %s */}",
        jsx_attribute = "// %s",
        comment = "// %s",
      },
    },
  },
}

return options
```
- plugins/init.lua
```lua
  {
    "nvim-treesitter/nvim-treesitter",
    cmd = { "TSInstall", "TSBufEnable", "TSBufDisable", "TSModuleInfo" },
    build = ":TSUpdate",
    opts = function()
      return require "configs.treesitter"
    end,
    config = function(_, opts)
      dofile(vim.g.base46_cache .. "syntax")
      require("nvim-treesitter.configs").setup(opts)
    end,
  },
```

**Comment by siduck** [comment]
so it works now i   suppose? 

btw you dont need write the full table, for example the table u wrote is our default treesitter table, and then you want to   override its config func, then u  do this 

```lua
 {  
    "treesitter blabla",
     config = function(_, opts)
      dofile(vim.g.base46_cache .. "syntax")
      require("nvim-treesitter.configs").setup(opts)
      
      Add your stuff here
    end ,
  }
```
    
- that config spec is a function so u have to include its contents too as it doesnt merges. 
    
 -   Look at the plugin examples here https://nvchad.com/docs/config/plugins#manage_plugins
    
  -  So in a nutshell lazy uses some sort of function to merge parent tables ( i.e from default nvchad list which is in nvchad repo ) and then your config 
    
-  Similar to `vim.tbl_deep_extend` https://nvchad.com/docs/config/walkthrough

**Comment by siduck** [comment]
so in your case as you're just overriding opts

you  use a table

```lua
 {
        "blabla tresiter ", 
        opts = { 
        
           ADD YOUR TABLE HERE 
           
           }
           
}

**Comment by siduck** [comment]
or you could save that table in a file, ( return it ) and in opts `opts = require("configs.treesitter")`
